Automattic/simplenote-ios
# iOS Widgets Final PR

### Fix
After many months, this PR encompasses the entire iOS widgets feature.  Fixes #1289 

All of the changes in this branch have been tested before, but never hurts to do a final pass. There are a few things to check here.

Widgets:
There are three widgets.  New note widget, Note Widget, and List Widget.
The new note widget creates a button that gives you a quick home screen button to create a new note

The note widget displays a preview of the content from a selected note

The list widget shows the top few notes from a given tag (or all notes)

### Test
New Note Widget (for more details about this widget check out #1293 ):
1. create a new note widget
2. tap on the button to make sure a new note is created.

Note Widget (for more details about this widget check out #1375 ):
1. add the note widget.  By default the note at the top of your notes list should appear
2. choose a note to display, make sure it is showing the same content (checkboxes will not render yet #1388 )
3. tap on the widget and you will be taken into the app and that note will load

List Widget (for more details about this widget check out #1389 ):
1. add the list widget.  by default all notes will be the tag that is selceted
2. choose a tag to display, once selected tap on the title of the tag to load into the app showing the tag list
3. return to the home screen and tap on one of the note titles, the note will load
4. tap on the new note button and see that a note is created tagged with the note the widget is displaying

### Review
***(Required)*** Add instructions for reviewers.  For example:
> Only one developer is are required to review these changes, but anyone can perform the review.

### Release
***(Required)*** Add a concise statement to `RELEASE-NOTES.txt` if the changes should be included in release notes. Include details about updating the notes in this section. For example:
> `RELEASE-NOTES.txt` was updated in 646382 with:
> 
> > - Added home screen widgets #1416
